$NetBSD$

Ensure interim libraries can be found.  These patches are only appropriate for
pkgsrc, other builds aren't as strict about removing $ORIGIN as us.

--- make/Images.gmk.orig	2026-10-30 12:27:06.000000000 +0000
+++ make/Images.gmk
@@ -171,7 +171,7 @@ define CreateCDSArchive
       DEPS := $$(jlink_jdk), \
       OUTPUT_FILE := $$(JDK_IMAGE_DIR)/$$($1_$2_CDS_ARCHIVE), \
       SUPPORT_DIR := $$(JDK_IMAGE_SUPPORT_DIR), \
-      COMMAND := $$(FIXPATH) $$(JDK_IMAGE_DIR)/bin/java -Xshare:dump \
+      COMMAND := $$(FIXPATH) LD_LIBRARY_PATH=$$(BUILD_JDK)/lib:$$(BUILD_JDK)/lib/server $$(JDK_IMAGE_DIR)/bin/java -Xshare:dump \
           -XX:SharedArchiveFile=$$(JDK_IMAGE_DIR)/$$($1_$2_CDS_ARCHIVE) \
           -$1 $$($1_$2_DUMP_EXTRA_ARG) $$($1_$2_CDS_DUMP_FLAGS) $$(LOG_INFO), \
   ))
@@ -184,7 +184,7 @@ define CreateCDSArchive
       DEPS := $$(jlink_jre), \
       OUTPUT_FILE := $$(JRE_IMAGE_DIR)/$$($1_$2_CDS_ARCHIVE), \
       SUPPORT_DIR := $$(JRE_IMAGE_SUPPORT_DIR), \
-      COMMAND := $$(FIXPATH) $$(JRE_IMAGE_DIR)/bin/java -Xshare:dump \
+      COMMAND := $$(FIXPATH) LD_LIBRARY_PATH=$$(BUILD_JDK)/lib:$$(BUILD_JDK)/lib/server $$(JRE_IMAGE_DIR)/bin/java -Xshare:dump \
           -XX:SharedArchiveFile=$$(JRE_IMAGE_DIR)/$$($1_$2_CDS_ARCHIVE) \
           -$1 $$($1_$2_DUMP_EXTRA_ARG) $$($1_$2_CDS_DUMP_FLAGS) $$(LOG_INFO), \
   ))
